tiledb/sm/storage_manager/consolidator.h

```diff
@@ -137,49 +294,165 @@ class Consolidator {
-  Status create_subarray(Array* array, void** subarray) const;
+  /**
+   * Computes the union of the non-empty domains of `fragments` between
+   * `start` and `end`. Note that this set of fragments must contain at
+   * least one dense fragment.
+   *
+   * @tparam T The domain type.
+   * @param array_schema The array schema.
+   * @param fragments The fragments input.
+   * @param start The function will compute the union of the non-empty
+   *    domains focusing only of fragments between `start` and `end`.
+   * @param end The function will compute the union of the non-empty
+   *    domains focusing only of fragments between `start` and `end`.
+   * @param uinion_non_empty_domains The union of the non-empty domains
+   *    to compute.
+   * @param coincides This is set to `true` by the function if the
+   *     computed union of non-empty domains coincides with the array space
+   *     tiles.
+   * @return Status
+   *
+   * @note  This function is not applicable to all sparse fragments.
+   */
+  template <
+      class T,
+      typename std::enable_if<std::is_integral<T>::value, T>::type* = nullptr>
+  Status compute_union_non_empty_domains(
+      const ArraySchema* array_schema,
+      const std::vector<FragmentInfo>& fragments,
+      size_t start,
+      size_t end,
+      T* union_non_empty_domains,
+      bool* coincides) const {
+    // Non-applicable to all sparse fragments
+    if (all_sparse(fragments, start, end))
+      return Status::Ok();
+
+    auto dim_num = array_schema->dim_num();
+    auto domain_size = 2 * array_schema->coords_size();
+    auto domain = (T*)array_schema->domain()->domain();
+    auto tile_extents = (T*)array_schema->domain()->tile_extents();
+    assert(
+        !fragments.empty() && start < fragments.size() &&
+        end < fragments.size() && start <= end);
+
+    // Initialize subarray to the first fragment non-empty domain
+    std::memcpy(
+        union_non_empty_domains,
+        fragments[start].non_empty_domain_,
+        domain_size);
+
+    // Initialize subarray
+    for (size_t i = start + 1; i <= end; ++i) {
+      utils::geometry::expand_mbr_with_mbr<T>(
+          union_non_empty_domains,
+          (const T*)fragments[i].non_empty_domain_,
+          dim_num);
+    }
+
+    *coincides = true;
+    for (unsigned i = 0; i < dim_num; ++i) {
+      assert(
+          (union_non_empty_domains[2 * i + 1] - domain[2 * i]) !=
+          std::numeric_limits<T>::max());
+      if (((union_non_empty_domains[2 * i] - domain[2 * i]) %
+           tile_extents[i]) ||
+          (union_non_empty_domains[2 * i + 1] - domain[2 * i] + 1) %
+              tile_extents[i]) {
+        *coincides = false;
+        break;
+      }
+    }
+
+    return Status::Ok();
+  }
+
+  /** Non-applicable for float/double domains. */
+  template <
+      class T,
+      typename std::enable_if<!std::is_integral<T>::value, T>::type* = nullptr>
+  Status compute_union_non_empty_domains(
```

Comment:
Similarly to another case in `utils.h`, I don't think this second definition is necessary.

Reply:
This case is similar to `cell_num` discussed above.